src/V3Inline.cpp

```diff
@@ -296,30 +296,32 @@ class InlineRelinkVisitor final : public VNVisitor {
                 // remove the change detection on the output variable.
                 UINFO(9, "public pin assign: " << exprvarrefp << endl);
                 UASSERT_OBJ(!nodep->isNonOutput(), nodep, "Outputs only - inputs use AssignAlias");
-                m_modp->addStmtsp(
-                    new AstAssignW{flp, new AstVarRef{flp, exprvarrefp->varp(), VAccess::WRITE},
-                                   new AstVarRef{flp, nodep, VAccess::READ}});
+                m_modp->addStmtsp(new AstAssignW{flp, exprvarrefp->cloneTree(false),
```

Comment:
Please also add an assert for `exprvarrefp->access() == VAccess::WRITE` (or clone and force to VAccess::WRITE), unless you can prove this is always correct as is (I can't).

Reply:
The assert for `nodep` being an output port is just above and variable references passed to output ports are marked as `WRITE`.